cluster/gce/addons/apiserver-network-proxy/daemonset.yaml

```diff
@@ -0,0 +1,57 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
```

Comment:
Why is this a daemonset vs a deployment?

Reply:
Its not clear that every node needs to have an agent running on it, I do not want more than 1 agent running on a node.

Reply:
we probably want a few (e.g., log(N)) of these per node pool. daemonset is fine for starters.

Avoiding more than one per node is easily accomplished with a bunch of work involving taints, affinity rules, and anti-affinity rules... 🙃